lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -959,16 +1003,30 @@ function dosomething_reportback_get_reportback_files_query($params = array()) {
   $query->join('users', 'u', 'rb.uid = u.uid');
   $query->join('node', 'n', 'rb.nid = n.nid');
   $query->join('file_managed', 'f', 'rbf.fid = f.fid');
+
   if (isset($params['tid'])) {
     $query->join('field_data_field_primary_cause', 't', 't.entity_id = n.nid');
     $query->condition('t.field_primary_cause_tid', $params['tid']);
     $query->condition('t.entity_type', 'node');
   }
+
   if (isset($params['status'])) {
     $query->condition('rbf.status', $params['status']);
   }
+
   if (isset($params['nid'])) {
-    $query->condition('rb.nid', $params['nid']);
+    if (is_array($params['nid'])) {
+      $query->condition('rb.nid', $params['nid'], 'IN');
+    } else {
+      $query->condition('rb.nid', $params['nid']);
```

Comment:
Yeah, I've tried it a bit before and it doesn't fail... just give unexpected results a bit, since it defaults to grabbing items without any filter on it (for like items for a specific nid). No failure per se.